executor/window.go

```diff
@@ -174,3 +175,130 @@ func (e *WindowExec) copyChk(chk *chunk.Chunk) {
 		chk.MakeRefTo(i, childResult, col.Index)
 	}
 }
+
+// windowProcessor is the interface for processing different kinds of windows.
+type windowProcessor interface {
```

Comment:
`consumeGroupRows` can be removed?
The argument `rows` in `appendResult2Chunk` contains all rows in a same partition, so we can do aggregation at there once instead of calling `consumeGroupRows` multiple times?

Reply:
In https://github.com/pingcap/tidb/pull/9358/files#diff-509643ad918102b310cd4be4c55fc644R198, we actually remove rows that already processed, this can result in less memory usage.